python/ray/serve/tests/unit/test_metrics_utils.py

```diff
@@ -21,159 +59,149 @@ def inc():
             nonlocal val
             val += 1
 
-        MetricsPusher.NO_TASKS_REGISTERED_INTERVAL_S = 0.01
-        metrics_pusher = MetricsPusher()
+        metrics_pusher = MetricsPusher(get_or_create_event_loop())
         metrics_pusher.start()
-        assert len(metrics_pusher.tasks) == 0
-        assert metrics_pusher.pusher_thread.is_alive()
+        assert len(metrics_pusher._tasks) == 0
 
         metrics_pusher.register_or_update_task("inc", inc, 0.01)
 
-        wait_for_condition(lambda: val > 0, timeout=10)
+        async_wait_for_condition(lambda: val > 0, timeout=10)
 
-    def test_basic(self):
-        start = 0
-        timer = MockTimer(start)
+    @pytest.mark.asyncio
+    async def test_basic(self):
+        timer = MockAsyncTimer(0)
 
-        with patch("time.time", new=timer.time), patch(
-            "time.sleep", new=timer.realistic_sleep
-        ):
-            counter = {"val": 0}
-            result = {}
-            expected_result = 20
+        with patch("ray.serve._private.metrics_utils.async_sleep", timer.sleep):
```

Comment:
What's the reason for `patch`ing this in? Can we instead dependency-inject this as an argument to the `MetricsPusher`? Then you can avoid the `import asyncio.sleep as ...` confusion too.

Reply:
I've changed it to dependency injection!